torch/utils/data/dataset.py

```diff
@@ -206,6 +208,54 @@ def __len__(self):
         return self.tensors[0].size(0)
 
 
+class StackDataset(Dataset[Union[Tuple[T_co, ...], Dict[str, T_co]]]):
+    r"""Dataset as a stacking of multiple datasets.
+
+    This class is useful to assemble different parts of complex input data, given as datasets.
+
+    Example:
+        >>> # xdoctest: +SKIP
+        >>> images = ImageDataset()
+        >>> texts = TextDataset()
+        >>> tuple_stack = StackDataset(images, texts)
+        >>> tuple_stack[0] == (images[0], texts[0])
+        >>> dict_stack = StackDataset(image=images, text=texts)
+        >>> dict_stack[0] == {'image': images[0], 'text': texts[0]}
+
+    Args:
+        *args (Dataset): Datasets for stacking returned as tuple.
+        **kwargs (Dataset): Datasets for stacking returned as dict.
+    """
+    datasets: Union[Tuple[Dataset[T_co], ...], Dict[str, Dataset[T_co]]]
+
+    def __init__(self, *args: Dataset[T_co], **kwargs: Dataset[T_co]) -> None:
+        if args:
+            if kwargs:
+                raise ValueError("Supported only tuple or dict like output setup, but both given")
+            if any(len(args[0]) != len(dataset) for dataset in args):  # type: ignore[arg-type]
+                raise ValueError("Size mismatch between datasets")
+
+            self._length = len(args[0])  # type: ignore[arg-type]
+            self.datasets = args
+        elif kwargs:
+            tmp = list(kwargs.values())
+            if any(len(tmp[0]) != len(dataset) for dataset in tmp):  # type: ignore[arg-type]
+                raise ValueError("Size mismatch between datasets")
+
+            self._length = len(tmp[0])  # type: ignore[arg-type]
+            self.datasets = kwargs
+        else:
+            raise ValueError("At least one dataset should be passed")
+
+    def __getitem__(self, index):
```

Comment:
Need a type annotation of TypeVar here

Reply:
but output already typed in generic Dataset.